### PR TITLE
Handle communication errors nicely

### PR DESCRIPTION
**Pull request by arnelandwehr**
- used wiremock as chaotic proxy (not so great but we will have
      toxiproxy in the real ITs)
    - restuctured the code to have a more concise result handling (before
      modification error were handled by checked exceptions
    - added the abbility to configure purchase timeouts
    - introduced a new test builder (which is still highly redundant and has
      to be cleaned up in furhter PRs). Was required as i had to configure
      the adyen uri.
    - warnings cleaned up

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentMethodInfoPlugin.java [hunk 1]
```diff
@@ -16,12 +16,12 @@
 
 package org.killbill.billing.plugin.adyen.api;
 
-import java.util.UUID;
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentMethodInfoPlugin.java]: Extra shuffling - can you remove these changes from the PR for clarity?

Comment (arnelandwehr) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentMethodInfoPlugin.java]: not sure what the goal is. Normally imports are added/removed and sorted by the IDE. With checked in code style (as discussed in the chat) there should be no change if the code was properly formatted (which is sadly not the case here).

I could remove the change but the next person will experience the same problem. An alternative would be to do a mass format as explicit commit. 

It's no fun to contribute if you always have to fear that pressing "format all" will kill every diff :(

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 2]
```diff
@@ -270,29 +277,31 @@ public PaymentTransactionInfoPlugin creditPayment(final UUID kbAccountId, final
         throw new PaymentPluginApiException(null, "CREDIT: unsupported operation");
     }
 
+    // HPP
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Refund is not HPP, only `buildFormDescriptor` below is.

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 3]
```diff
@@ -185,39 +188,41 @@ protected String getPaymentMethodId(final AdyenPaymentMethodsRecord paymentMetho
...
-                                         },
-                                         kbAccountId,
-                                         kbPaymentId,
-                                         kbTransactionId,
-                                         kbPaymentMethodId,
-                                         amount,
-                                         currency,
-                                         properties,
-                                         context);
+                new TransactionExecutor<PurchaseResult>() {
```
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Similarly, it doesn't look like there is a functional change here (only indentation & materialization of `AdyenPaymentServiceProviderPort`)? Can we tackle them in another PR?

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentMethodInfoPlugin.java (hunk 1 above)
Comment (pierre) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentMethodInfoPlugin.java]: >  there should be no change if the code was properly formatted (which is sadly not the case here).

I just reformatted the file in my IDE and it didn't trigger any change (the [import ordering](https://github.com/killbill/killbill/blob/master/.idea/codeStyleSettings.xml#L14-L29) looks good, no?).

> It's no fun to contribute if you always have to fear that pressing "format all" will kill every diff :(

But it's no fun either to review a PR with 23 changed files, 1,595 additions and 798 deletions, especially for a bug fix that tricky. I'm having a hard time understanding functionally what changed.

**Pull request closed without merging**